### PR TITLE
feat: Kosmosctl supports knode join and unjoin

### LINKED ISSUE #70: Kosmosctl needs to support join and unjoin cluster to knode-manager
**What would you like to be added**:
kosmosctl support join and unjoin cluster to knode-manager, like join cluster to  multi-cluster network module, e.g: 
kosmosctl join -f member-cluster.yaml --master-kubeconfig=[master-kubeconfig] --cluster-kubeconfig=[member-kubeconfig]

The addition of knode may be as follows: 
kosmosctl join -f knode.yaml --master-kubeconfig=[master-kubeconfig]
image

**Why is this needed**:
We need a way for users to easily and quickly add clusters to our multi-cluster management and orchestration module, which will help improve user experience.

### PR DESCRIPTION
**Pull request by ONE7live**
#### What type of PR is this?

**feat:** Kosmosctl supports knode join and unjoin
#### What does this PR do?

Support a way for users to easily and quickly add clusters to our multi-cluster management and orchestration module.
#### Which issue(s) does this PR fix?

Fixes #70
#### Special notes for your reviewer:

/cc @xyz2277
/cc @wuyingjun-lucky
#### Screenshots (if applicable):

```release-note

```

#### Additional Comments:

```release-note

```

### CODE REVIEW

File: pkg/kosmosctl/install/install.go [hunk 1]
```diff
@@ -185,118 +209,146 @@ func (o *CommandInstallOptions) runClusterlink() error {
...
+func (o *CommandInstallOptions) runClustertree() error {
+	klog.Info("Start creating kosmos-clustertree...")
 	namespace := &corev1.Namespace{}
 	namespace.Name = o.Namespace
-	_, err := o.Client.CoreV1().Namespaces().Create(context.TODO(), namespace, metav1.CreateOptions{})
-	if err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			return fmt.Errorf("kosmosctl install clusterrouter run error, namespace options failed: %v", err)
+	_, err := o.Client.CoreV1().Namespaces().Get(context.TODO(), util.DefaultNamespace, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
```
Comment (xyz2277) [inline pkg/kosmosctl/install/install.go]: if err is not nill and IsNotFound(err),  there is not code for this situation.

File: pkg/kosmosctl/install/install.go [hunk 2]
```diff
@@ -185,118 +209,146 @@ func (o *CommandInstallOptions) runClusterlink() error {
...
+			return fmt.Errorf("kosmosctl install clustertree run error, serviceaccount options failed: %v", err)
 		}
 	}
+	klog.Info("ServiceAccount clustertree-knode-manager has been created.")
 
-	clusterRouterServiceAccount, err := util.GenerateServiceAccount(manifest.ClusterRouterKnodeServiceAccount, manifest.ServiceAccountReplace{
-		Namespace: o.Namespace,
-	})
+	klog.Info("Start creating kosmos-clustertree ClusterRole...")
+	clusterTreeClusterRole, err := util.GenerateClusterRole(manifest.ClusterTreeKnodeManagerClusterRole, nil)
```
Comment (xyz2277) [inline pkg/kosmosctl/install/install.go]: clusterTree should be clustertree, clustertree is a word

File: pkg/kosmosctl/unjoin/unjoin.go [hunk 3]
```diff
@@ -123,37 +161,55 @@ func (o *CommandUnJoinOptions) Run(f ctlutil.Factory, cmd *cobra.Command, args [
...
+	}
+
+	klog.Info("Cluster [" + clusterName + "] was removed.")
+	return nil
+}
+
+func (o *CommandUnJoinOptions) runKnode(knodeName string) error {
+	klog.Info("Start removing knode from kosmos control plane...")
+	for {
+		err := o.DynamicClient.Resource(util.KnodeGVR).Namespace("").Delete(context.TODO(), knodeName, metav1.DeleteOptions{})
```
Comment (xyz2277) [inline pkg/kosmosctl/unjoin/unjoin.go:203]: Is namespace "" ok?

Comment (ONE7live) [reply pkg/kosmosctl/unjoin/unjoin.go:203]: It has been tested here and the process can proceed normally,  ns is not specified when the knode is created

File: pkg/kosmosctl/install/install.go (hunk 2 above)
Comment (ONE7live) [reply pkg/kosmosctl/install/install.go]: ok, i'm done

File: pkg/kosmosctl/install/install.go (hunk 1 above)
Comment (ONE7live) [reply pkg/kosmosctl/install/install.go]: I've modified the logic here, is it possible now?

**Pull request merged**